tsdb/agent/db_test.go

```diff
@@ -109,15 +110,16 @@ func TestCommit(t *testing.T) {
 	reg = prometheus.NewRegistry()
 	remoteStorage = remote.NewStorage(log.With(logger, "component", "remote"), reg, startTime, promAgentDir, time.Second*30, nil)
 
-	s, err = Open(logger, nil, remoteStorage, promAgentDir, opts)
-	if err != nil {
-		t.Fatalf("unable to create storage for the agent: %v", err)
-	}
+	s1, err := Open(logger, nil, remoteStorage, promAgentDir, opts)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, s1.Close())
+	}()
```

Comment:
I'm not actually sure why the tests here were re-opening the WAL (they came from grafana#22 and I didn't notice that this was happening initially). I ran into this in #9623 and removed the duplicate opens. 

Doing the same here might help remove the need for the `defer func()`, maybe?

Reply:
(that is, I think a lot of the problem is that these tests are opening the DB twice when they don't need to)

Reply:
@spaparaju Do you happen to remember why the tests were doing this?

Reply:
@rfratto I see some disconnect here. Currently at the Prometheus main branch, I could not see the agent tests (`/tsdb/agent/storage_test.go` file is missing) that I included as part of these two PRs ([first](https://github.com/grafana/prometheus/pull/22/files), [second](https://github.com/grafana/prometheus/pull/24/files)).